src/sentry/release_health/release_monitor/metrics.py

```diff
@@ -0,0 +1,152 @@
+import logging
+import time
+from collections import defaultdict
+from datetime import datetime, timedelta
+from typing import Mapping, Sequence
+
+from snuba_sdk import Column, Condition, Direction, Entity, Granularity, Op, OrderBy, Query
+
+from sentry.release_health.release_monitor.base import BaseReleaseMonitorBackend, Totals
+from sentry.sentry_metrics.indexer.strings import SESSION_METRIC_NAMES
+from sentry.snuba.dataset import Dataset, EntityKey
+from sentry.utils import metrics
+from sentry.utils.snuba import raw_snql_query
+
+logger = logging.getLogger(__name__)
+
+
+class MetricReleaseMonitorBackend(BaseReleaseMonitorBackend):
+    def fetch_projects_with_recent_sessions(self) -> Mapping[int, Sequence[int]]:
+        with metrics.timer(
+            "release_monitor.fetch_projects_with_recent_sessions.loop", sample_rate=1.0
+        ):
+            aggregated_projects = defaultdict(list)
+            start_time = time.time()
+            offset = 0
+            while (time.time() - start_time) < self.MAX_SECONDS:
+                query = (
```

Comment:
So re this query I am not sure it makes sense to add the service layer.. maybe we could extend it but this is not really a data query so I am not sure if it logically makes sense

Reply:
Let me think about this use case for a while